examples.md

```diff
@@ -62,6 +63,28 @@ steps:
       ${{ runner.os }}-go-
 ```
 
+## Haskell - Cabal
+
+We cache the elements of the Cabal store separately, as the entirety of `~/.cabal` can grow very large for projects with many dependencies.
+
+```yaml
+- uses: actions/cache@v1
+  name: Cache ~/.cabal/packages
+  with:
+    path: ~/.cabal/packages
+    key: ${{ runner.os }}-${{ matrix.ghc }}-cabal-packages
+- uses: actions/cache@v1
+  name: Cache ~/.cabal/store
+  with:
+    path: ~/.cabal/store
+    key: ${{ runner.os }}-${{ matrix.ghc }}-cabal-store
+- uses: actions/cache@v1
+  name: Cache dist-newstyle
+  with:
+    path: dist-newstyle
```

Comment:
Caching this folder is also of modest benefit; it mostly helps for large projects whose local artifacts (i.e. not the build-deps) are costly to build, otherwise you risk accumulating garbage and wasting the limited cache folder space (also you might risk running into subtle corruption issues when updating `cabal` as the `dist-newstyle` folder isn't properly versioned); this is why I intentionally avoided this folder being included in #38 as well as in general for the `dist-newstyle` folder.

Reply:
This is super helpful information and I think definitely work a comment in the example docs as to the tradeoffs.

We could also add cabal version to the cache key for to help with the issue of updating `cabal` for projects where caching `dist-newstyle` is appropriate.